sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```diff
@@ -44,58 +44,45 @@ private MarathonConstraintParser() {
     }
 
     /**
-     * ANDs the provided marathon-style constraint string onto the provided hard-coded
-     * {@link PlacementRule}, or returns the provided {@link PlacementRule} as-is if the
-     * marathon-style constraint is {@code null} or empty.
-     *
-     * @throws IOException if {@code marathonConstraints} couldn't be parsed, or if the parsed
-     *     content isn't valid or supported
-     */
-    public static PlacementRule parseWith(PlacementRule rule, String marathonConstraints)
-            throws IOException {
-        PlacementRule marathonRule = parse(marathonConstraints);
-        if (marathonRule instanceof PassthroughRule) {
-            return rule; // pass-through original rule
-        }
-        return new AndRule(rule, marathonRule);
-    }
-
-    /**
      * Creates and returns a new {@link PlacementRule} against the provided marathon-style
      * constraint string. Returns a {@link PassthroughRule} if the provided constraint string is
      * {@code null} or empty.
      *
+     *
+     * @param podName
```

Comment:
Needs javadoc content. Can't hurt to specify what to do if a given constraint string is to be exposed across the whole service and not just a single pod, or if that's even supported..